aerogear/data-sync-ui
# add ability for UI to be standalone

# Motivaton

This change will allow us to address some technical debt and architecture limitations.
Most important ones:
- Add ability to try out queries and schemas without publishing them to the server
- Add ability to Query schema when the server is protected
- Separate UI from Server (server will be production environment and UI will be development environment)
- Add support for additional use cases (user provided sync server can use our execution later)
- Improve and fix schema reload for sync
- Unify models that are being used in UI and Server (so they will be extended in one single place)
- Support schema seeding (memeo and other schemas - this is currently happening on Server - we need that ability on UI)

TL;DR - This is a crucial change that fixes some problems with the Sync architecture and it will give us tons of benefits with very little investment.

# Side effects of this change

- UI and Server will reuse the same module
- Development time will require using github version of new module (or releases to npm)
When we will get more and more modules we can consider moving towards mono repositories etc. but currently this impact is marginal

# Verification Steps

To verify that these changes work, it can be done as part of the overall verification of the feature. Using this repository and the sync UI.

1. Use docker-compose to create the db locally. 
1. Run the server with `npm run dev:keymemeo`
1. Run the UI with `npm run server`
1. Everything should work as expected with the server and the UI. i.e. authentication should be required on the server and not on the UI.
